Tests/CatalogItems/Set-RsSubscription.Tests.ps1

```diff
@@ -0,0 +1,346 @@
+        Description           = "Shared on \\unc\path"
+        EventType             = "TimedSubscription"
+        IsDataDriven          = $false
+	    MatchData             = $matchData.OuterXml
+    }
+    
+    return $subscription
+}
+
+Describe "Set-RsSubscription" {
+        Context "Set-RsSubscription without parameters"{
+                $folderName = 'SutGetRsItemReference_MinParameters' + [guid]::NewGuid()
+                New-RsFolder -Path / -FolderName $folderName
+                $folderPath = '/' + $folderName
+                $localResourcesPath =   (Get-Item -Path ".\").FullName  + '\Tests\CatalogItems\testResources\emptyReport.rdl'
+                Write-RsCatalogItem -Path $localResourcesPath -RsFolder $folderPath
+                $report = (Get-RsFolderContent -RsFolder $folderPath )| Where-Object TypeName -eq 'Report'
+                $subscription = Get-NewFileShareSubscription
+
+                $localResourcesPath =   (Get-Item -Path ".\").FullName  + '\Tests\CatalogItems\testResources\UnDataset.rsd'
+                Write-RsCatalogItem -Path $localResourcesPath -RsFolder $folderPath
+                $dataSet = (Get-RsFolderContent -RsFolder $folderPath ) | Where-Object TypeName -eq 'DataSet'
+                $DataSetPath = $folderPath + '/UnDataSet'
+                
+                $newRSDSName = "DataSource"
+                $newRSDSExtension = "SQL"
+                $newRSDSConnectionString = "Initial Catalog=DB; Data Source=Instance"
+                $newRSDSCredentialRetrieval = "Store"
+                #Dummy credentials
+                $Pass = ConvertTo-SecureString -String "123" -AsPlainText -Force
+                $newRSDSCredential = New-Object -TypeName System.Management.Automation.PSCredential -ArgumentList "sql", $Pass
+                New-RsDataSource -RsFolder $folderPath -Name $newRSDSName -Extension $newRSDSExtension -ConnectionString $newRSDSConnectionString -CredentialRetrieval $newRSDSCredentialRetrieval -DatasourceCredentials $newRSDSCredential
+
+                $DataSourcePath = "$folderPath/$newRSDSName"
+
+                $RsDataSet = Get-RsItemReference -Path $report.Path | Where-Object ReferenceType -eq 'DataSet'
+                $RsDataSource = Get-RsItemReference -Path $report.Path | Where-Object ReferenceType -eq 'DataSource'
+                $RsDataSetSource = Get-RsItemReference -Path $DataSetPath | Where-Object ReferenceType -eq 'DataSource'
+
+                #Set data source and data set for all objects
+                Set-RsDataSourceReference -Path $DataSetPath -DataSourceName $RsDataSetSource.Name -DataSourcePath $DataSourcePath
+                Set-RsDataSourceReference -Path $report.Path -DataSourceName $RsDataSource.Name -DataSourcePath $DataSourcePath
+                Set-RsDataSetReference -Path $report.Path -DataSetName $RsDataSet.Name -DataSetPath $dataSet.Path
+                
+
+                Set-RsSubscription -Subscription $subscription -Path $report.Path
```

Comment:
if you have extra time add a test case for the piping, set-rssubscription seems to be more useful in the case of piping as there is not a pshell function in the suite to create subscriptions (New-RsSubscription)